### ISSUE TITLE
interaction with LaTeX?

### ISSUE THREAD

**Issue by nbecker** [amperser/proselint#277]
Does proselint grok LaTeX?  I'm guessing for now converting LaTeX -> plain txt is best?

**Comment by suchow** [comment]
It doesn't yet handle LaTeX source code in any special way, but that won't stop it from running over the document. You may find that it falsely flags some of the markup (e.g., straight quotes in image tags), but otherwise it should be fine.

**Comment by j6k4m8** [comment]
I'm interested in extending this to run in LaTeX .tex documents. Do you have a preference for how/where I implement? Is this a `plugin`?

**Comment by nbecker** [comment]
I have no preference

**Comment by suchow** [comment]
@j6k4m8 The narrowest way to handle this (recommended) is to create a new topic detector and then exclude that new topic for all the checks that false alarm on LaTeX syntax. The ultimate solution is to parse the text from the LaTeX and apply Proselint to that parsed representation.

https://github.com/amperser/proselint/blob/master/proselint/tools.py#L433

**Comment by j6k4m8** [comment]
@suchow Got it. My plan was to parse the LaTeX, keeping track of line-number so that the output could still reflect the original line-numbers and character positions. I imagine this would require modification to:

| Code | Description |
|-------|------------|
| [`main.`CLI args](https://github.com/amperser/proselint/blob/master/proselint/command_line.py#L113) | Adding a `--latex` arg (unless you want to infer from the text or file ext?)
| [`tools.line_and_column`](https://github.com/amperser/proselint/blob/master/proselint/tools.py#L198) | Modification to enable line number "hints" instead of using the enumerated line number from iterating over the file |

With a super basic python script to strip out LaTeX `\keywords{}`, such as this...

```python
import sys
import re


for i, line in enumerate(sys.stdin):
    i += 1
    line = re.sub(re.compile(r"\\\w+\{(.*?)\}"), r"\1\n", line)

    line = [n.strip() for n in line.split("\n") if n.strip()]
    for l in line:
        print("{}\t{}".format(i, l))
```

...I think we can have a working model pretty quickly. I appreciate there are [already tools that do this](https://github.com/pkubowicz/opendetex) but I'm not sure to what extent we can rely on them, and I'm not sure how they perform with the issue I mention below.

I think it's important to strip these tags out and parse the split text separately because of how commonly folks include bracketed section headers in the text content:

```latex
\section{Section Title} Here is the content of my
section. As you can see, lorem ipsum dolor sit amet...
```

i.e. "Section Title" should be parsed separately from "Here is the...". Not _entirely_ sure how we prevent splitting sentences on non-linebreak tokens like `\textit{}` yet (especially on user-defined symbols, vs builtins). Thoughts welcome.

**Comment by Foadsf** [comment]
one method which I found useful is to convert latex to plain text with [tex2txt](tex2txt) or markdown with [pandoc](https://pandoc.org/):


    pandoc -f latex -t markdown main.tex > main.md

and then use proselint.

**Comment by nzt** [comment]
You can use proselint in latex with [textlint](https://textlint.github.io) + [textlint-rule-proselint](https://github.com/textlint-rule/textlint-rule-proselint) + [textlint-plugin-latex2e](https://github.com/textlint/textlint-plugin-latex2e)
